src/main/kotlin/com/github/imflog/schema/registry/parser/AvroSchemaParser.kt

```diff
@@ -17,8 +22,45 @@ class AvroSchemaParser(
         schemaContent: String,
         localReferences: List<LocalReference>
     ): String {
-        val parser = Schema.Parser()
-        localReferences.map { reference -> reference.content(rootDir) }.reversed().forEach { parser.parse(it) }
-        return parser.parse(schemaContent).toString()
+        val jsonObj = JSONObject(schemaContent)
+        if(jsonObj.has(FIELDS)) {
+            jsonObj.put(FIELDS,
+                JSONArray(jsonObj
+                    .getJSONArray(FIELDS)
+                    .toMutableList()
+                    .mapInPlace { if (it is JSONObject) replaceLocalReference(it, localReferences) else it }
+                )
+            )
+        }
+        return jsonObj.toString()
+    }
+
+    fun replaceLocalReference(jsonObject: JSONObject, localReferences: List<LocalReference>): JSONObject {
```

Comment:
I think this does not work with `array` (the type is in the `items` field) nor for `map` (the type is in the `values` field).

Reply:
You are right, I will add these cases